server/ui.go

```diff
@@ -50,19 +48,37 @@ func newUIOption(frontendAddr string, uiIP string, uiPort int, configDir string)
 	return WithUI(uiserver.NewServer(uiserveroptions.WithConfigProvider(cfg))), nil
 }
 
-func NewUIConfig(frontendAddr string, uiIP string, uiPort int, configDir string) (*uiconfig.Config, error) {
+func NewUIConfig(c *uiConfig, configDir string) (*uiconfig.Config, error) {
 	cfg := &uiconfig.Config{
-		Host: uiIP,
-		Port: uiPort,
+		Host:                c.Host,
+		Port:                c.Port,
+		TemporalGRPCAddress: c.TemporalGRPCAddress,
+		EnableUI:            c.EnableUI,
+		Codec: uiconfig.Codec{
+			Endpoint: c.CodecEndpoint,
+		},
 	}
+
 	if configDir != "" {
 		if err := provider.Load(configDir, cfg, "temporal-ui"); err != nil {
 			if !strings.HasPrefix(err.Error(), "no config files found") {
 				return nil, err
 			}
 		}
 	}
-	cfg.TemporalGRPCAddress = frontendAddr
-	cfg.EnableUI = true
+
+	// See https://github.com/temporalio/temporalite/pull/174/files#r1035958308
+	// for context about those overrides.
+	cfg.TemporalGRPCAddress = c.TemporalGRPCAddress
+	cfg.EnableUI = c.EnableUI
+
 	return cfg, nil
 }
+
+type uiConfig struct {
```

Comment:
Why not use `uiconfig.Config` instead of creating this new struct and renaming `NewUIConfig` to `MergeWithConfigFile`?
This is a bit confusing to me.

Reply:
good point